src/Concordium/Client/Runner.hs

```diff
@@ -1507,32 +1586,33 @@ processAccountCmd action baseCfgDir verbose backend =
 
       input <- case inputMaybe of
         Nothing -> do
-          logInfo [[i|the ACCOUNT argument was not provided; using the default account name '#{defaultAccountName}'|]]
-          return defaultAccountName
+          case Map.lookup defaultAccountName (bcAccountNameMap baseCfg) of
+            Nothing -> do
+              logFatal [[i|The ACCOUNT argument was not provided; so the default account name '#{defaultAccountName}' was used, but no account with that name exists.|]]
+            Just acc -> do
+              logInfo [[i|The ACCOUNT argument was not provided; so the default account name '#{defaultAccountName}' was used.|]]
+              return . Text.pack $ show acc
         Just acc -> return acc
 
-      accountIdentifier <- case Types.decodeAccountIdentifier (Text.encodeUtf8 input) of
-                Just _ -> return input -- input is a wellformed account identifier
-                Nothing -> case Map.lookup input (bcAccountNameMap baseCfg) of
-                    Just a -> return $ Text.pack $ show a -- input is the local name of an account
-                    Nothing -> if isNothing inputMaybe
-                               then logFatal [[i|The ACCOUNT argument was not provided; so the default account name '#{defaultAccountName}' was used, but no account with that name exists.|]]
-                               else logFatal [[i|The identifier '#{input}' is neither a credential registration ID, an account index, the address nor the name of an account|]]
+      accountIdentifier <-
+        case Types.decodeAccountIdentifier (Text.encodeUtf8 input) of
+          Just v -> return v -- input is a wellformed account identifier
+          Nothing -> logFatal [[i|The identifier '#{input}' is neither a credential registration ID, an account index, the address nor the name of an account|]]
 
       (accInfo, na, dec) <- withClient backend $ do
         -- query account
-        (bbh, accInfoValue :: GRPCResult Value) <- withBestBlockHash block (\bbh -> (bbh,) <$> getAccountInfo accountIdentifier bbh)
-        accInfo <- do 
-          case accInfoValue of
-            Left err -> logFatal [err]
-            Right aiv -> case AE.fromJSON $ grpcResponseVal aiv of
-                            AE.Error err -> logFatal ["Cannot decode account info response from the node: " ++ err]
-                            AE.Success Nothing -> logFatal [printf "Account %s does not exist on the chain." $ show accountIdentifier]
-                            AE.Success (Just air) -> return air
+        bhInput <- readBlockHashOrDefault Best block
+        accInfo <- do
+          res <- getAccountInfoV2 accountIdentifier bhInput
+          dieOnRequestFailed ["I/O error:"] res
+          dieOnStatusNotFound [[i|Account does not exist on the chain.|]] res
+          dieOnConversionError ["Cannot decode account info response from the node:"] res
+          getResponseValueOrDie res
```

Comment:
Is there a reason for not using `getAccountInfoOrDie` here?

Reply:
Yea, due to the signature, but that's changed in the next PR.